admin/plugins.php

```diff
@@ -390,3 +458,87 @@ function perflab_render_plugin_card( array $plugin_data ) {
+
+/**
+ * Handles the standalone plugin install and activation via AJAX.
+ *
+ * @since n.e.x.t
+ */
+function perflab_install_activate_standalone_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid nonce: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! isset( $_REQUEST['data'] ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid plugin data: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! function_exists( 'plugins_api' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	}
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	$plugins_to_install = $_REQUEST['data'];
+	$plugins_to_install = explode( ',', $plugins_to_install );
+
+	$get_standalone_plugins_module_map = perflab_get_standalone_plugins_module_map();
+	$get_standalone_plugins_file_map   = perflab_get_standalone_plugins_file_map();
+	$modules                           = perflab_get_module_settings();
+
+	$status = array();
+	foreach ( $plugins_to_install as $plugin_slug ) {
+
+		$api = perflab_query_plugin_info( $plugin_slug );
+
+		$module_slug = isset( $get_standalone_plugins_module_map[ $plugin_slug ] ) ? $get_standalone_plugins_module_map[ $plugin_slug ] : '';
+		$plugin_slug = $get_standalone_plugins_file_map[ $plugin_slug ];
+
+		// Check if the plugin is already installed.
+		if ( ! is_plugin_installed( $plugin_slug ) ) {
+			$plugin_path = WP_PLUGIN_DIR . '/' . $plugin_slug;
+
+			// Check if the plugin zip file exists.
+			if ( ! file_exists( $plugin_path ) ) {
+
+				// Replace new Plugin_Installer_Skin with new Quiet_Upgrader_Skin when output needs to be suppressed.
+				$skin     = new WP_Ajax_Upgrader_Skin( array( 'api' => $api ) );
+				$upgrader = new Plugin_Upgrader( $skin );
+				$result   = $upgrader->install( $api['download_link'] );
+
+				if ( is_wp_error( $result ) ) {
+					$status['errorMessage'] = $result->get_error_message();
+					wp_send_json_error( $status );
+				} elseif ( is_wp_error( $skin->result ) ) {
+					$status['errorMessage'] = $skin->result->get_error_message();
+					wp_send_json_error( $status );
+				} elseif ( $skin->get_errors()->has_errors() ) {
+					$status['errorMessage'] = $skin->get_error_messages();
+					wp_send_json_error( $status );
+				}
+			}
+		}
+
+		$result = activate_plugin( $plugin_slug );
+		if ( is_wp_error( $result ) ) {
+			$status['errorMessage'] = $result->get_error_message();
+			wp_send_json_error( $status );
+		}
+
+		// Deactivate legacy modules.
+		foreach ( $modules as $slug => $data ) {
+			if ( $module_slug === $slug ) {
+				unset( $modules[ $slug ] );
+			}
+		}
```

Comment:
I think this loop is unnecessary.
```suggestion
		unset( $modules[ $module_slug ] );
```

Reply:
😔